src/main/java/org/folio/note/NoteRepositoryImpl.java

```diff
@@ -0,0 +1,205 @@
+      .save(NOTE_TABLE, note.getId(), note, future.completer());
+
+    return future.map(noteId -> {
+      note.setId(noteId);
+      return note;
+    });
+  }
+
+  /**
+   * Fetches a note record from the database
+   *
+   * @param id id of note to get
+   */
+  @Override
+  public Future<Note> getOneNote(String id, String tenantId) {
+    Future<Note> future = Future.future();
+    PostgresClient.getInstance(vertx, tenantId)
+      .getById(NOTE_VIEW, id, NoteView.class,
+        reply -> {
+          if (reply.succeeded()) {
+            NoteView noteView = reply.result();
+            if (Objects.isNull(noteView)) {
+              future.fail(new NotFoundException("Note " + id + " not found"));
+            } else {
+              future.complete(mapNoteView(noteView));
+            }
+          } else {
+            String error = PgExceptionUtil.badRequestMessage(reply.cause());
+            logger.error(error, reply.cause());
+            if (error == null) {
+              future.fail(reply.cause());
+            } else {
+              future.fail(new BadRequestException(reply.cause().getMessage()));
+            }
+          }
+        });
+    return future;
+  }
+
+  @Override
+  public Future<Void> deleteNote(String id, String tenantId) {
+    Future<UpdateResult> future = Future.future();
+    PostgresClient.getInstance(vertx, tenantId)
+      .delete(NOTE_TABLE, id, future.completer());
+    return future.map(updateResult -> {
+      if(updateResult.getUpdated() == 0){
+        throw new NotFoundException();
+      }
+      return null;
+    });
+  }
+
+  @Override
+  public Future<Void> updateNote(String id, Note note, String tenantId) {
+    Future<UpdateResult> future = Future.future();
+    if (note.getLinks().isEmpty()) {
+      PostgresClient.getInstance(vertx, tenantId)
+        .delete(NOTE_TABLE, id, future.completer());
+    } else {
+      PostgresClient.getInstance(vertx, tenantId)
+        .update(NOTE_TABLE, note, id, future.completer());
+    }
+    return future
+      .map(updateResult -> {
+        if(updateResult.getUpdated() == 0){
+          throw new NotFoundException();
+        }
+        return (Void) null;
+      })
+      .recover(throwable -> {
+        String badRequestMessage = PgExceptionUtil.badRequestMessage(throwable);
+        if (badRequestMessage != null) {
+          return failedFuture(new BadRequestException(badRequestMessage, throwable));
+        } else {
+          return failedFuture(throwable);
+        }
+      });
+  }
+
+  private Future<NoteCollection> getNotes(String tenantId, CQLWrapper cql) {
+    Future<Results<NoteView>> future = Future.future();
+    PostgresClient.getInstance(vertx, tenantId)
+      .get(NOTE_VIEW, NoteView.class, new String[]{"*"}, cql,
+        true /*get count too*/, false /* set id */,
+        future.completer());
+
+    return future
+      .map(this::mapNoteResults);
+  }
+
+  private NoteCollection mapNoteResults(Results<NoteView> results) {
```

Comment:
I think if you can read directly into Note objects then the code can be simplified